baremaps-workflow/src/main/java/org/apache/baremaps/workflow/Step.java

```diff
@@ -14,4 +14,11 @@
 
 import java.util.List;
 
+/**
+ * A step is a group of tasks executed sequentially.
```

Comment:
Maybe in this comment we should also mention that a step is part of a DAG to make the `needs` parameter more understandable?

Reply:
I improved the documentation in the workflow and step classes.